compiler/src/dmd/globals.d

```diff
@@ -170,6 +170,7 @@ extern (C++) struct Param
     bool addMain;           // add a default main() function
     bool allInst;           // generate code for all template instantiations
     bool bitfields;         // support C style bit fields
+    bool sarif = false;     // track if the --sarif flag is passed
```

Comment:
if you are adding variables here, you will need to update the corresponding header, `globals.h`

Reply:
Sure, but @dkorpel suggested avoiding compiler flags. If that's the approach I should take, then `globals.h` might not need changes either. Please let me know which direction I should move forward with

Reply:
Since it seems to work with output on `stderr`, it would probably be most useful if you can manage to make SARIF a [`MessageStyle`](https://github.com/dlang/dmd/blob/master/compiler/src/dmd/location.d#L27), since then you can use `-verror-style=[digitalmars|gnu|sarif]` and output all the compiler output as JSON.

This might be complicated by `pragma(msg)` a bit (the output is printed on stdout), but that can come later.

Reply:
done- https://github.com/dlang/dmd/pull/16986/commits/19f15d6e8712693be161eb82d96bb542d63c2948